### DISCUSSION TITLE
Any example with Locate?

### DISCUSSION THREAD

**Discussion by woile** [winnow-rs/winnow#492] [Q&A] [answered]
Hi there! Thanks for this awesome library. I was wondering if we could add an example showing how to locate row and column in case of error. I see most of the info points to [line_span](https://docs.rs/line-span/latest/line_span/), which is fine, but I think it would be better to have a simple example. Thoughts?

I still don't know how to do it, but if there's no one else, I could give it a try if it's welcome.

Regards

**Comment by epage** [comment]
I'm ok with adding an example to `Located`.  We can also point to the implementation of [translate_position](https://github.com/winnow-rs/winnow/blob/85d49337298eba5ade9d0cd5a498436b29f052c0/src/error.rs#L1263) as a hand-implementation of how to do it.

**Reply by woile** [reply]
How is it generally approached? Is it better to keep track of the line and col while processing or just a position, and on error calculate the line and col?

**Reply by epage** [reply]
imo line,column is a UI concern so conversion to it should be reserved for when rendering something for the user.

**Reply by woile** [reply]
I made a small example with `Located`... and I'm not sure what is it for. 

Looking at the error output, the offset is already there 🤔 and the `translate_position` can give you the `line` and `col`, and `line_span` can give beginning and end of the error line.

**Reply by epage** [reply, marked as answer]
`Located` is more for when using `Parser::with_span`
